### PR TITLE
ARM template for Azure EventHub function

### PR DESCRIPTION
**Pull request by himanshu219**

### CODE REVIEW

File: logs_build/EventHubs/function.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (duchatran) [inline logs_build/EventHubs/function.json]: Since we call the folder EventHubs_Metrics, shouldn't we call this EventHubs_Logs?

File: logs_build/EventHubs/sumometricclient.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
+/*jshint esversion: 6 */
```
Comment (duchatran) [inline logs_build/EventHubs/sumometricclient.js]: As this is for logs, maybe it's better to remove this file to avoid confusion

File: metrics_build/EventHubs_Metrics/sumoclient.js [hunk 3]
```diff
@@ -0,0 +1,262 @@
+/*jshint esversion: 6 */
```
Comment (duchatran) [inline metrics_build/EventHubs_Metrics/sumoclient.js]: Is this file needed for metrics collection? If not maybe we can remove.

File: logs_build/EventHubs/sumoclient.js [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+
+/**
+ * Default method to generate a headersObj object for the bucket
+ * @param message input message
+ */
+SumoClient.prototype.generateHeaders = function(message) {
+    let sourceCategory = (this.options.metadata)? (this.options.metadata.category || '') :'';
+    let sourceName = (this.options.metadata)? (this.options.metadata.name || ''):'' ;
+    let sourceHost = (this.options.metadata)? (this.options.metadata.host || ''):'';
+    let headerObj = {'X-Sumo-Name':sourceName, 'X-Sumo-Category':sourceCategory, 'X-Sumo-Host':sourceHost, 'X-Sumo-Client': 'eventhublogs-azure-function'};
```
Comment (duchatran) [inline logs_build/EventHubs/sumoclient.js]: I like the X-Sumo-Client header!

File: tests/activity_log_fixtures.json [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+                "http://schemas.microsoft.com/claims/authnclassreference":"1","aio":"Y2NgYDgv1K/hPiFd+27SQn2lsKBIz8+rvPUk1y/1femdVL7390kA",
+                "http://schemas.microsoft.com/claims/authnmethodsreferences":"pwd",
+                "appid":"f64071b9-a79b-4655-9dad-3b3535e00b84","appidacr":"2","e_exp":"263099","http://schemas.xmlsoap.org/ws/2005/05/identity/claims/surname":"Pal","http://schemas.xmlsoap.org/ws/2005/05/identity/claims/givenname":"Himanshu","ipaddr":"111.93.54.106","name":"Himanshu Pal","http://schemas.microsoft.com/identity/claims/objectidentifier":"95fdefa0-a7b8-47a3-b969-046862c9a979","puid":"10030000A77E725E","http://schemas.microsoft.com/identity/claims/scope":"user_impersonation","http://schemas.xmlsoap.org/ws/2005/05/identity/claims/nameidentifier":"mucf8YYRxalNePxQtx2t_Vp7WtIYsjrE68K5r0JCo9k","http://schemas.microsoft.com/identity/claims/tenantid":"a39bedba-be8f-4c0f-bfe2-b8c7913501ea","http://schemas.xmlsoap.org/ws/2005/05/identity/claims/name":"himanshu@itsumologic.onmicrosoft.com","http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn":"himanshu@itsumologic.onmicrosoft.com","uti":"kINdgmA35kmIMsXXinoXAA","ver":"1.0"
```
Comment (duchatran) [inline tests/activity_log_fixtures.json]: Is there anything sensitive (credit, key) here?

File: logs_build/EventHubs/sumometricclient.js (hunk 2 above)
Comment (himanshu219) [reply logs_build/EventHubs/sumometricclient.js]: I also thought so first but in mainindex.js both sumometricclient and sumoclient are included.

File: metrics_build/EventHubs_Metrics/sumoclient.js (hunk 3 above)
Comment (himanshu219) [reply metrics_build/EventHubs_Metrics/sumoclient.js]: nops it is not needed but in mainindex it's there as a dependency

File: tests/activity_log_fixtures.json (hunk 5 above)
Comment (himanshu219) [reply tests/activity_log_fixtures.json]: not any keys but I guess I should remove my account name himanshu@itsumologic.onmicrosoft.com from there

File: logs_build/EventHubs/function.json (hunk 1 above)
Comment (himanshu219) [reply logs_build/EventHubs/function.json]: done
